File: c++/009.cpp
// Copyright 2012 Bob Wilkinson <bob@fourtheye.org>

// This is a solution for http://projecteuler.net/problem=8

#include <cstdio>

int main() {
  int a, b, c;
  int stop = 1000;
  for (a = 1; a < stop; a++) {
    for (b = a + 1; b < stop; b++) {
      for (c = b + 1; c < stop; c++) {
        if ((a + b +c) == stop) {
          if ((a * a) + (b * b) == (c * c)) {
            printf("(%d, %d, %d) -> %d\n", a, b, c, a*b*c);
            return(0);
          }
        }
      }
    }
  }
  return(1);
}

File: python/009.py
#!/usr/bin/env python

def main():
  """ this computes projecteuler.net problem 009 """

  stop = 1000
  for a in range(1,stop):
    for b in range(a+1,stop):
      for c in range(b+1,stop):
        if (a+b+c == stop):
          if ((a*a)+(b*b) == (c*c)):
            print a*b*c
            exit(0)
  exit(1)

if __name__ == "__main__":
    main()
